### PR TITLE
[Woo POS] Payment error retry handling

### PR DESCRIPTION
**Pull request by joshheald**
Part of: #13481 
Merge after: #13576

## Description

This PR adds support for showing `Try another payment method` vs `Try payment again` on retryable errors of different types. 

I've also simplified the `CardPresentPaymentEventDetails.paymentError` case so it now covers retryable and non-retryable errors. That will make it clearer to read in the client code, to see when we get an error.

Not covered:
 - making the CTA on non-retryable errors actually work (you're a bit stuck on that screen)
 - making the Exit order button on retryable orders work (it will only show for Try again errors)

## Steps to reproduce

Edit a POS-eligible product to have a sale price ending in `.01` – [see error amounts here](https://docs.stripe.com/terminal/references/testing#physical-test-cards)

### Using another payment method
1. Lauch the app and go to POS
2. Add your error product to the cart
3. Check out
4. Tap your card
5. Observe that you're shown a retryable `Try another payment method` error

### Using the same payment method
1. Launch the app and go to POS
2. Add a non-error product to the cart
3. Check out
4. Tap your card, and immediately disable network connectivity on the iPad
5. Observe that you're shown a retryable `Try payment again` error
6. Re-enable the network
7. Tap `Retry`
8. Observe that payment is taken without tapping the card again.

## Screenshots

https://github.com/user-attachments/assets/32d893f0-85bb-40e7-90ac-ee8a5085012c

---
- [x] I have considered adding unit tests for this change. If I decided not to add them, I have provided a brief explanation below (optional):
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.
- [ ] This PR includes refactoring; smoke testing of the entire section is needed.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/POS/Presentation/Card Present Payments/Reader Messages/PointOfSaleCardPresentPaymentErrorMessageViewModel.swift [hunk 1]
```diff
@@ -35,13 +49,20 @@ private extension PointOfSaleCardPresentPaymentErrorMessageViewModel {
             comment: "Error message. Presented to users after collecting a payment fails on the Point of Sale Checkout"
         )
 
-        static let tryAgain =  NSLocalizedString(
+        static let tryPaymentAgain =  NSLocalizedString(
             "pointOfSale.cardPresent.paymentError.tryAgain.button.title",
```
Comment (iamgabrielma) [inline WooCommerce/Classes/POS/Presentation/Card Present Payments/Reader Messages/PointOfSaleCardPresentPaymentErrorMessageViewModel.swift]: Nit:
```suggestion
            "pointOfSale.cardPresent.paymentError.tryPaymentAgain.button.title",
```

**Review by iamgabrielma — COMMENTED** [review]
Code looks good! I'll wrap up testing later today as I cannot test the second case on a simulator (context: p1723535353508709-slack-C025A8VV728 )

**Review by iamgabrielma — APPROVED** [review]
LGTM :shipit:

**Pull request merged**